patching/plugin-commands-patching/src/stateFile.ts

```diff
@@ -0,0 +1,77 @@
+import fs from 'fs'
+import path from 'path'
+import util from 'util'
+
+export type StateKey = string
+
+export interface StateValue {
+  selector: string
+  applyToAll: boolean
```

Comment:
might be better to use a range of versions here. But not sure about it. Whether we need that complexity.

Reply:
I think we can add it later when we actually need it. This state file is not exposed as a public API, and would be reset to `{}` (empty object) after all `patch-commit`, so the effects of the design decision we made here is minimal.

Also, `applyToAll` here let `patch-commit` knows whether the user has answered "yes" or "no" to "Apply this patch to all versions?" during `pnpm patch`.